local_planner/src/nodes/planner_functions.cpp

```diff
@@ -248,14 +244,20 @@ double costFunction(int e, int z, const nav_msgs::GridCells& path_waypoints,
   double cost;
   int waypoint_index = path_waypoints.cells.size();
 
-  double dist = (position - goal).norm();
-  double dist_old = (position_old - goal).norm();
-  Eigen::Vector3f candidate_goal =
-      fromPolarToCartesian(e, z, dist, toPoint(position));
+  float dist = (position - goal).norm();
+  float dist_old = (position_old - goal).norm();
+  PolarPoint p_pol = {};
+  p_pol.e = static_cast<float>(e);
```

Comment:
no need to cast anymore

Reply:
you could also make to cost function take a polar point (reference) as an input. That way you would have less arguments.